airflow/providers/databricks/plugins/databricks_workflow.py

```diff
@@ -0,0 +1,453 @@
+            TaskInstance.dag_id == dag_id,
+            TaskInstance.run_id == dag_run.run_id,
+            TaskInstance.task_id == operator.task_id,
+        )
+        .one_or_none()
+    )
+    if not ti:
+        raise TaskInstanceNotFound("Task instance not found")
+    return ti
+
+
+def get_xcom_result(
+    ti_key: TaskInstanceKey,
+    key: str,
+    ti: TaskInstance | None,
+) -> Any:
+    result = XCom.get_value(
+        ti_key=ti_key,
+        key=key,
+    )
+    from airflow.providers.databricks.operators.databricks_workflow import WorkflowRunMetadata
+
+    return WorkflowRunMetadata(**result)
+
+
+class WorkflowJobRunLink(BaseOperatorLink, LoggingMixin):
+    """Constructs a link to monitor a Databricks Job Run."""
+
+    name = "See Databricks Job Run"
+
+    def get_link(
+        self,
+        operator: BaseOperator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        ti = None
+        if not ti_key:
+            ti = get_task_instance(operator, dttm)
+            ti_key = ti.key
+        task_group = operator.task_group
+
+        if not task_group:
+            raise AirflowException("Task group is required for generating Databricks Workflow Job Run Link.")
+
+        dag = airflow_app.dag_bag.get_dag(ti_key.dag_id)
+        dag.get_task(ti_key.task_id)
+        self.log.info("Getting link for task %s", ti_key.task_id)
+        if ".launch" not in ti_key.task_id:
+            self.log.debug("Finding the launch task for job run metadata %s", ti_key.task_id)
+            launch_task_id = get_launch_task_id(task_group)
+            ti_key = _get_launch_task_key(ti_key, task_id=launch_task_id)
+        # Should we catch the exception here if there is no return value?
+        metadata = get_xcom_result(ti_key, "return_value", ti)
+
+        hook = DatabricksHook(metadata.conn_id)
+        return f"https://{hook.host}/#job/{metadata.job_id}/run/{metadata.run_id}"
+
+
+class WorkflowJobRepairAllFailedLink(BaseOperatorLink, LoggingMixin):
+    """Constructs a link to send a request to repair all failed tasks in the Databricks workflow."""
+
+    name = "Repair All Failed Tasks"
+
+    def get_link(
+        self,
+        operator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        ti = None
+        if not ti_key:
+            ti = get_task_instance(operator, dttm)
+            ti_key = ti.key
+        task_group = operator.task_group
+        self.log.debug(
+            "Creating link to repair all tasks for databricks job run %s",
+            task_group.group_id,
+        )
+        # Should we catch the exception here if there is no return value?
+        metadata = get_xcom_result(ti_key, "return_value", ti)
+
+        tasks_str = self.get_tasks_to_run(ti_key, operator, self.log)
+        self.log.debug("tasks to rerun: %s", tasks_str)
+        return (
+            f"/repair_databricks_job?dag_id={ti_key.dag_id}&"
```

Comment:
This won't work if the Airflow is not rooted at `/`, we'll need to use Flask's `url_for` helpers.

Reply:
using Flask's url_for now, thanks.